server/src/main/java/org/apache/druid/client/coordinator/CoordinatorClient.java

```diff
@@ -101,4 +103,13 @@ public interface CoordinatorClient
    * API: {@code POST /druid/coordinator/v1/config}
    */
   ListenableFuture<Void> updateCoordinatorDynamicConfig(CoordinatorDynamicConfig dynamicConfig);
+
+  /**
+   * Gets the lookup configuration for a tier.
+   * <p>
+   * API: {@code GET /druid/coordinator/v1/lookups/config/<tier>}
+   *
+   * @param tier The name of the tier for which the lookup configuration is to be fetched.
+   */
+  ListenableFuture<Map<String, LookupExtractorFactoryContainer>> fetchLookupsForTier(String tier);
```

Comment:
Please rename method to signify that this is a synchronous method unlike the others in this interface.
Since it is synchronous, we need not return a future.

```suggestion
  Map<String, LookupExtractorFactoryContainer> fetchLookupsForTierSync(String tier);
```

Reply:
makes sense, I was planning to do that however i wondered if it makes sense to make everything async to maintain consistency in the client (thought it'll server no purpose)

changing the signature for now

Reply:
Yes, we need not rename the other methods since they are expected to be async.